File: java/src/test/java/com/dsa/examples/unionfind/UnionFindTest.java
package com.dsa.examples.unionfind;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class UnionFindTest {

    @Test
    void unionAndFind() {
        UnionFind uf = new UnionFind(5);
        assertEquals(5, uf.countComponents());
        uf.union(0, 1);
        uf.union(2, 3);
        assertTrue(uf.connected(0, 1));
        assertFalse(uf.connected(0, 2));
        uf.union(1, 2);
        assertTrue(uf.connected(0, 3));
        assertEquals(2, uf.countComponents());
    }

    @Test
    void singleElement() {
        UnionFind uf = new UnionFind(1);
        assertTrue(uf.connected(0, 0));
    }
}

File: javascript/test/unionfind/unionFind.test.js
const { UnionFind } = require('../../src/unionfind/unionFind');

describe('UnionFind', () => {
  test('initial components', () => {
    const uf = new UnionFind(5);
    expect(uf.countComponents()).toBe(5);
    expect(uf.connected(0, 1)).toBe(false);
  });

  test('union and connected', () => {
    const uf = new UnionFind(5);
    uf.union(0, 1);
    uf.union(2, 3);
    expect(uf.connected(0, 1)).toBe(true);
    expect(uf.countComponents()).toBe(3);
    uf.union(1, 2);
    expect(uf.connected(0, 3)).toBe(true);
  });

  test('redundant union', () => {
    const uf = new UnionFind(3);
    uf.union(0, 1);
    uf.union(0, 1);
    expect(uf.countComponents()).toBe(2);
  });
});
